### ISSUE TITLE
True Positive detections

### ISSUE THREAD

**Issue by ilpapds** [ultralytics/yolov5#8440]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello, i am trying to store/save the true positive (correct) detected bounding boxes of the yolov5 model. In order to do that, i am trying to modify the val.py code in the following lines:

            # Evaluate
            if nl:
                tbox = xywh2xyxy(labels[:, 1:5])  # target boxes
                scale_coords(im[si].shape[1:], tbox, shape, shapes[si][1])  # native-space labels
                labelsn = torch.cat((labels[:, 0:1], tbox), 1)  # native-space labels
                correct = process_batch(predn, labelsn, iouv)
                if plots:
                    confusion_matrix.process_batch(predn, labelsn)
            stats.append((correct, pred[:, 4], pred[:, 5], labels[:, 0]))  # (correct, conf, pcls, tcls)

Modyfying them in this way:

            # Evaluate
            if nl:
                tbox = xywh2xyxy(labels[:, 1:5])  # target boxes
                scale_coords(im[si].shape[1:], tbox, shape, shapes[si][1])  # native-space labels
                labelsn = torch.cat((labels[:, 0:1], tbox), 1)  # native-space labels
                correct = process_batch(predn, labelsn, iouv)
                for batch_j in correct[batch_j, 0]:  
                    init_img = cv2.imread(paths[batch_j])                               
                    crop_img = init_img[int(predn[batch_j,0]):int(predn[batch_j,2]),int(predn[batch_j,1]):int(predn[batch_j,3]),:]     
                if plots:
                    confusion_matrix.process_batch(predn, labelsn)
            stats.append((correct, pred[:, 4], pred[:, 5], labels[:, 0]))  # (correct, conf, pcls, tcls)

And now i want to store this crop_img which corresponds to the true positive detections. Is that possible? Has anybody done it with another way? Thank you mery much

**Comment by glenn-jocher** [comment]
@ilpapds 👋 Hello! Thanks for asking about **cropping results** with YOLOv5 🚀.  Cropping bounding box detections can be useful for training classification models on box contents for example. This feature was added in PR https://github.com/ultralytics/yolov5/pull/2827. You can crop detections using either detect.py or YOLOv5 PyTorch Hub:

## detect.py
Crops will be saved under `runs/detect/exp/crops`, with a directory for each class detected.
```bash
python detect.py --save-crop
```
<img width="450" alt="Original" src="https://user-images.githubusercontent.com/26833433/139577918-65cc8f5d-741d-4a88-8753-0154013c624d.jpg">

<img width="1067" alt="Crop" src="https://user-images.githubusercontent.com/26833433/139577896-aa26b07c-466d-4057-9bfe-b01b56007587.png">

## YOLOv5 PyTorch Hub
Crops will be saved under `runs/detect/exp/crops` if `save=True`, and also returned as a dictionary with crops as numpy arrays.
```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # or yolov5m, yolov5l, yolov5x, custom

# Images
img = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, PIL, OpenCV, numpy, list

# Inference
results = model(img)

# Results
crops = results.crop(save=True)  # or .show(), .save(), .print(), .pandas(), etc.
```

Good luck 🍀 and let us know if you have any other questions!

**Comment by ilpapds** [comment]
@glenn-jocher thank you for your quick response. However, instead of cropping the detected bounding boxes, i wish to filter only the true positive ones. This is why i am trying to use the val.py code. Could you please help me on how to achieve this? Thank you

**Comment by glenn-jocher** [comment]
@ilpapds sorry, we don't have resources to review or assist with custom code, but perhaps you can use the above examples to help build your custom solution.
